### PR TITLE
feat(ga): add a new datasource to get the list of the endpoint groups

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new datasource to get the list of the endpoint groups.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new datasource to get the list of the endpoint groups.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_group)$ make testacc TEST="./huaweicloud/services/acceptance/ga" TESTARGS="-run TestAccDatasourceEndpointGroups_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ga -v -run TestAccDatasourceEndpointGroups_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEndpointGroups_basic
=== PAUSE TestAccDatasourceEndpointGroups_basic
=== CONT  TestAccDatasourceEndpointGroups_basic
--- PASS: TestAccDatasourceEndpointGroups_basic (126.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ga        126.560s
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (dev_group)$ 
```

### CODE REVIEW

File: docs/data-sources/ga_endpoint_groups.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+subcategory: "GA (Global Accelerator)"
```
Comment (deer-hang) [inline docs/data-sources/ga_endpoint_groups.md]: The subcategory in resource is `subcategory: "Global Accelerator (GA)"`.
Is there any special reason for writing it this way?

Comment (ruwenqiang123) [reply docs/data-sources/ga_endpoint_groups.md]: done

**Pull request merged**